tools/twix/src/twix_painter.rs

```diff
@@ -179,22 +249,39 @@ impl<Frame> TwixPainter<Frame> {
     }
 
     pub fn rect_filled(&self, min: Point2<Frame>, max: Point2<Frame>, fill_color: Color32) {
-        let rect = Rect {
-            min: self.transform_world_to_pixel(min),
-            max: self.transform_world_to_pixel(max),
-        };
-        self.painter
-            .rect_filled(sort_rect(rect), Rounding::ZERO, fill_color);
+        let right_bottom = point![max.x(), min.y()];
+        let left_top = point![min.x(), max.y()];
+
+        let points: Vec<_> = vec![
+            self.transform_world_to_pixel(min),
+            self.transform_world_to_pixel(right_bottom),
+            self.transform_world_to_pixel(max),
+            self.transform_world_to_pixel(left_top),
+        ];
+
+        self.painter.add(Shape::Path(PathShape::convex_polygon(
+            points,
+            fill_color,
+            Stroke::default(),
+        )));
```

Comment:
The egui painter only supports axis aligned rectangles.
Since we now need to draw rotated rectangles, specifying just two corners is no longer sufficient to describe the whole shape.

Reply:
Ah, that makes sense, thanks!